src/gravity/gravity.cpp

```diff
@@ -56,51 +73,53 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
-    params.Add("x", pin->GetOrAddReal("gravity", "x", 0.0));
-    params.Add("y", pin->GetOrAddReal("gravity", "y", 0.0));
-    params.Add("z", pin->GetOrAddReal("gravity", "z", 0.0));
-
-    const Real gm = pin->GetReal("gravity", "gm");
-    const Real qbin = pin->GetReal("gravity", "q");
-    const Real abin = pin->GetReal("gravity", "a");
-    const Real ebin = pin->GetOrAddReal("gravity", "e", 0.0);
-    const Real ibin = pin->GetOrAddReal("gravity", "i", 0.0) * M_PI / 180.;
-    const Real obin = pin->GetOrAddReal("gravity", "omega", 0.0) * M_PI / 180.;
-    const Real Obin = pin->GetOrAddReal("gravity", "Omega", 0.0) * M_PI / 180.;
-    const Real fbin = pin->GetOrAddReal("gravity", "f", 180.0) * M_PI / 180.;
+
+    params.Add("soft1", pin->GetOrAddReal(block_name, "soft1", 0.0));
+    params.Add("soft2", pin->GetOrAddReal(block_name, "soft2", 0.0));
+    params.Add("sink1", pin->GetOrAddReal(block_name, "sink1", 0.0));
+    params.Add("sink2", pin->GetOrAddReal(block_name, "sink2", 0.0));
+    params.Add("sink_rate1", pin->GetOrAddReal(block_name, "sink_rate1", 0.0));
+    params.Add("sink_rate2", pin->GetOrAddReal(block_name, "sink_rate2", 0.0));
+    params.Add("x", pin->GetOrAddReal(block_name, "x", 0.0));
+    params.Add("y", pin->GetOrAddReal(block_name, "y", 0.0));
+    params.Add("z", pin->GetOrAddReal(block_name, "z", 0.0));
+
+    const Real qbin = pin->GetReal(block_name, "q");
+    const Real abin = pin->GetReal(block_name, "a");
+    const Real ebin = pin->GetOrAddReal(block_name, "e", 0.0);
+    const Real ibin = pin->GetOrAddReal(block_name, "i", 0.0) * M_PI / 180.;
+    const Real obin = pin->GetOrAddReal(block_name, "omega", 0.0) * M_PI / 180.;
+    const Real Obin = pin->GetOrAddReal(block_name, "Omega", 0.0) * M_PI / 180.;
+    const Real fbin = pin->GetOrAddReal(block_name, "f", 180.0) * M_PI / 180.;
     Orbit orb(gm, abin, ebin, ibin, obin, Obin, fbin);
-    params.Add("gm", gm);
     params.Add("q", qbin);
     params.Add("orb", orb);
-  } else if (grav_type == "nbody") {
+  }
+  if (pin->DoesBlockExist("gravity/nbody")) {
+    count++;
     gtype = GravityType::nbody;
-    params.Add("gm", pin->GetReal("gravity", "gm"));
+    block_name = "gravity/nbody";
     const bool do_nbody = pin->GetBoolean("physics", "nbody");
-    PARTHENON_REQUIRE(do_nbody,
-                      "You have gravity/type = nbody but not physics/nbody = true!");
-  } else {
-    PARTHENON_FAIL("Unknown gravity type");
+    PARTHENON_REQUIRE(do_nbody, "You have <gravity/nbody> but not physics/nbody = true!");
   }
-  params.Add("type", gtype);
 
-  // Time start/stop controllers for gravity?
-  const Real tstart =
-      pin->GetOrAddReal("gravity", "tstart", std::numeric_limits<Real>::lowest());
-  const Real tstop =
-      pin->GetOrAddReal("gravity", "tstop", std::numeric_limits<Real>::max());
-  params.Add("tstart", tstart);
-  params.Add("tstop", tstop);
+  PARTHENON_REQUIRE((gm != -Big<Real>()) && (needs_gm),
```

Comment:
and then `isnan`?

Reply:
yeah, i was avoiding the isnan call, but no reason not to i suppose

Reply:
done